src/asm/main.c

```diff
@@ -6,815 +6,96 @@
-
-	ptr = ft_strchr(str, COMMENT_CHAR);
-	if (ptr != NULL)
-		ptr[0] = 0;
-	ptr = ft_strchr(str, ';');
-	if (ptr != NULL)
-		ptr[0] = 0;
-	return (str);
-}
-
-int	count_matching_chars(char *str, char c)
-{
-	int i;
-	int count;
-
-	count = 0;
-	i = 0;
-	while (str[i])
-	{
-		if (str[i] == c)
-			count++;
-		i++;
-	}
-	return (count);
-}
-
-void destroy_char_array(char **array, int size)
-{
-	int i;
-
-	i = 0;
-	while (i < size)
-	{
-		if (array[i] != NULL)
-			free(array[i]);
-		i++;
-	}
-	free(array);
-	return ;
-}
-
-void	header_error(char *msg, t_asm *a)
-{
-	ft_printf("{red}ERROR:{eoc} %s\n", msg);
-	destroy_char_array(a->split, a->lines);
-	if (a->tmp != NULL)
-		free(a->tmp);
-	close(a->fd);
-	free(a);
-	exit(0);
-	return ;
-}
-
-void	validate_champion_name(t_asm *a, char *str)
-{
-	int		len;
-	char	*quote1;
-
-	if (a->have_name == 1)
-		return (header_error("file has multiple .name commands", a));
-	a->have_name = 1;
-	if (count_matching_chars(str, '"') != 2)
-		return (header_error(".name command not well formatted", a));
-	quote1 = ft_strchr(str, '"') + 1;
-	len = ft_strchr(quote1, '"') - quote1;
-	if (len < 1)
-		return (header_error(".name cannot be blank", a));
-	if (len > PROG_NAME_LENGTH)
-		return (header_error(".name length exceeds PROG_NAME_LENGTH", a));
-	ft_strncpy(a->header.prog_name, quote1, len);
-//	ft_printf("program name is [%s] ...\n", a->header.prog_name);	//comment out
-	a->valid_name = 1;
-	return ;
-}
-
-void	validate_champion_comment(t_asm *a, char *str)
-{
-	int		len;
-	char	*quote1;
-
-	if (a->have_comment == 1)
-		return (header_error("file has multiple .comment commands", a));
-	a->have_comment = 1;
-	if (count_matching_chars(str, '"') != 2)
-		return (header_error(".comment command not well formatted", a));
-	quote1 = ft_strchr(str, '"') + 1;
-	len = ft_strchr(quote1, '"') - quote1;
-	if (len < 1)
-		return (header_error(".comment cannot be blank", a));
-	if (len > COMMENT_LENGTH)
-		return (header_error(".comment length exceeds COMMENT_LENGTH", a));
-	ft_strncpy(a->header.comment, quote1, len);
-//	ft_printf("program comment is [%s] ...\n", a->header.comment);	//comment out
-	a->valid_comment = 1;
-	return ;
-}
-
-//go through array of strings
-//compare each first word to NAME_CMD_STRING and COMMENT_CMD_STRING
-//if first non-empty string is other than NAME or COMMENT, throw error
-//if multiple results, throw error
-//
-//make sure command has two quotation marks ("), otherwise error
-//make sure length between quotation marks is > 0 and < PROG_NAME_LENGHT and COMMENT_LENGTH
-//if not throw error
-//save address of first quotation mark + 1  as beginning of string, replace 2nd q mark with 0
-//   (or use strncpy)
-//   copy into output header
-
-
-void insert_into_char_array(t_asm *a, char *str)
-{
-	char	**new_array;
 	int		i;
+	char	*output_file;
 
-	new_array = ft_memalloc((a->lines + 1) * sizeof(char *));
-	i = 0;
-	while (i < a->lines)
+	i = ft_strlen(original);
+	output_file = ft_memalloc(i + 5);
```

Comment:
`i + sizeof(".cor")` maybe? eh, your call.

Reply:
added #define BINARY_EXTENSION ".cor" to asm.h,
changed +5 to ft_strlen(BINARY_EXTENSION) + 1